feat(voting): voting widget UI

e18r opened the pull request:
Fixes AutarkLabs/open-enterprise#1884

e18r committed:
feat(voting): voting widget UI

ottodevs commented:
I rebased latest `master` branch, as we changed some dependencies there

ottodevs commented on app/components/Content/Widget.js:
```diff
@@ -1,9 +1,9 @@
+import { Card, GU } from '@aragon/ui'
 import MarkdownPreview from '../Widget/Markdown/Preview'
-import Votes from './Votes'
+import Voting from '../Widget/Voting/Voting'
```
`/* super-opinionated-comment-start */`

The widget is named "Latest Votes" so the component should be named `LatestVotes` or just `Votes`, why complicate things when we can use that simple approach?

Now we need to use `Voting` to refer to the `Latest Votes` widget, that furthermore, collides with the `Voting` app name... I suggest taking the easier path for everybody, name components are they are labelled in the designs, same for buttons, or any other component that has a label.

`/* super-opinionated-comment-end */`

ottodevs commented on app/components/Content/Widget.js:
```diff
@@ -25,15 +25,15 @@ const StyledCard = styled(Card).attrs({
   margin-bottom: 30px;
   position: relative;
-  padding: 24px;
+  padding: ${2 * GU}px;
```
`24px` equals `3 * GU` are we sure we want to reduce the padding from the designs?, I would confirm first with Javi or Yalda

ottodevs commented on app/components/Content/Widget.js:
```diff
@@ -25,15 +25,15 @@ const StyledCard = styled(Card).attrs({
     return <MarkdownPreview content={data.content} />
   case TYPE_VOTING:
-    return <Votes data={data} />
+    return <Voting data={data} />
```
It does not make sense for me to see `Voting` here, that word refers to an action or verb, while `Votes` is a descriptive word and it is also the widget name and label

ottodevs commented on app/api-react.js:
```diff
@@ -1,36 +1,46 @@
+      data: {
+        content: '# hola, `mundo`!',
+      },
+      position: {},
```
Not using position will cause an error and break the app, since we enforced PropTypes shape, I recommend to define position now.

ottodevs replied:
Solved

ottodevs commented on app/api-react.js:
```diff
@@ -1,36 +1,46 @@
+          },
+        ]
+      },
+      position: {},
```
Same as the previous comment applies here

ottodevs replied:
Solved

ottodevs commented on app/components/Widget/Voting/Voting.js:
```diff
@@ -0,0 +1,238 @@
+import { useNetwork } from '@aragon/api-react'
+import {
+  VOTING_STATUS,
+  VOTING_STATUS_ONGOING,
```
Maybe we should use `Object.keys(VOTING_STATUS)` instead of being so redundant with the status definitions?

ottodevs commented on app/components/Widget/Voting/Voting.js:
```diff
@@ -0,0 +1,238 @@
+`
+const Vote = styled(Card)`
+  margin: ${GU}px;
+  padding: ${GU}px;
```
I'd rather to define the all the paddings in the parent widget or here, but not half there and half here, this could make the CSS a madness to maintain

ottodevs commented on app/components/Widget/Voting/Voting.js:
```diff
@@ -0,0 +1,238 @@
+  display: flex;
+`
+const Vote = styled(Card)`
+  margin: ${GU}px;
```
Same with margin unless we have a good reason for that

ottodevs commented on app/components/Widget/Voting/Voting.js:
```diff
@@ -0,0 +1,238 @@
+  display: flex;
+`
+
+const App = ({ address }) => {
```
This name is so confusing, can we use a better, more descriptive name for this component?

ottodevs commented on app/components/Widget/Voting/Voting.js:
```diff
@@ -0,0 +1,238 @@
+    Icon = IconClose
+    color = theme.negative
+  }
+  else if (status !== VOTING_STATUS_ONGOING){
```
```suggestion
  else if (status !== VOTING_STATUS_ONGOING) {
```

ottodevs commented on app/api-react.js:
```diff
@@ -1,36 +1,46 @@
+    {
+      type: TYPE_VOTING,
+      data: {
+        votes: [
```
I am still not sure which is the best place for the `votes` array, just exploring it to be in the root state, the benefits are related to building it from the `store` are better consuming with `useAragonApi` instead of passing by props, also we avoid redundancy in the remote case multiple `Votes` widgets are added

ottodevs replied:
Solved

ottodevs commented on app/components/Widget/Voting/Voting.js:
```diff
@@ -0,0 +1,238 @@
+}
+
+const VoteContainer = styled.div`
+  display: flex;
```
This style is not enough, the votes cards don't look well and need to wrap.

Look:

![image](https://user-images.githubusercontent.com/5030059/73055916-15c91800-3e8e-11ea-87a6-90ca55abf0a5.png)

Even the overflow is hidden, so the widget becomes unusable

ottodevs replied:
Solved partially (now wraps but still needs some care)

ottodevs commented on app/utils/constants.js:
```diff
@@ -5,3 +5,10 @@ export const VOTING_STATUS_REJECTED = 'VOTING_STATUS_REJECTED'
 export const VOTING_STATUS_PENDING_ENACTMENT = 'VOTING_STATUS_PENDING_ENACTMENT'
 export const VOTING_STATUS_ENACTED = 'VOTING_STATUS_ENACTED'
+export const VOTING_STATUS = {
+  VOTING_STATUS_ONGOING: '',
```
Check the comment about using `Object.keys(VOTING_STATUS)` here for the rest of statuses

ottodevs reviewed (changes requested):
Some styling issues and some state that needs to be accommodated to the store structure, also some opinionated comments, just for convenience

I will work into some early fixes, especially CSS for this, so you can continue later

I also rebased `master` branch, hence some stuff was probably broken because the updates

ottodevs committed:
fix: accommodate state structure and style for rebase

- Change stubbed-api state structure to match latest master
- Improve api-react stubbed, add useNetwork and usePath
- Insert Voting component into the layout and remove unnecesary data
prop
- Fix vote cards wrapping style

ottodevs commented:
I solved some minor issues, the style needs some re-check specially when using the widget in the second column:

![image](https://user-images.githubusercontent.com/5030059/73057016-b8829600-3e90-11ea-8fda-102eeeae0b3c.png)

ottodevs committed:
refactor(styles): adjust margins, padding, some cleanup on Votes
